packages/nextjs/app/configure/_components/DownloadContracts.tsx

```diff
@@ -0,0 +1,124 @@
+"use client";
+
+import { useProvider } from "@starknet-react/core";
+import { useState } from "react";
+import { useTargetNetwork } from "~~/hooks/scaffold-stark/useTargetNetwork";
+
+export default function DownloadContracts() {
+  const { targetNetwork } = useTargetNetwork();
+  const [symbol, setSymbol] = useState<"Eth" | "Strk">("Eth");
+
+  const { provider } = useProvider();
+  const [address, setAddress] = useState<string>("");
+
+  const handleDownload = async () => {
+    if (!address) return;
+
+    const [apiResponse, classHash] = await Promise.all([
+      provider.getClassAt(address),
+      provider.getClassHashAt(address),
+    ]);
+
+    const contractsData = {
+      [targetNetwork.network]: {
```

Comment:
we should generate the exact `.ts` file that APPENDS the dowloaded contract to the already existing contracts on `predeployedContracts.ts` . check [this](https://github.com/Quantum3-Labs/scaffold-stark-2/blob/86026df6f9b463ae2095a01f675397044137fa41/packages/snfoundry/scripts-ts/helpers/parse-deployments.ts#L78-L85) as reference, however this example is used on the `deployedContracts.ts` . 

The `predeployedContracts.ts` file downloaded should be : { ... existing data, new_data }.

Reply:
also update instructions, no need to append, rather just replace the `predeployedContracts.ts`

Reply:
the current format in `.json` cannot easily be appended into the `.ts` file

Reply:
i think it would be a litte complex if we support update `predeployedContracts.ts` file online like use ` { ... existing data, new_data }.`.
because predeployedContracts is a typescript file not json file.
which means that we need to compile and transform  this file to javascript. and btw we'll extract all the constant from this file, which means we also need to import more than two file online. 

 so I suggest  two way:
* let the user themself to do the contract relacement
* we need a new file like `onlinePredeployedContract.ts`, and merge `onlinePredeployedContract.ts` `predeployedContracts .ts` `deployedContracts.ts` those three file in the code.

any comment is welcome :) @0xquantum3labs 


TLDR; need to transform ts file and use the compl tool like `ts-morph`(online version babel)

Reply:
plan 2:

https://github.com/Quantum3-Labs/scaffold-stark-2/pull/259/files#diff-3f3873a1760af995681d03ae297ac7f9f1b0349f5a7f151020905a9d3b1b0c43R104-R105

Reply:
```
we just provide the user a new predeployedContracts.ts he can directly copy paste it into the existing one, without affecting the existing contracts. Then the user can refresh and it should be good to go
```

sir i have made a new file name `configDeployedContracts.ts` which is download from this page, and user can replace and refresh, and it should be ok then.

Reply:
sir updated, ptal~ @0xquantum3labs  feel free to review

:)